src/Core/System/Snippet/Files/SnippetFileLoader.php

```diff
@@ -32,8 +38,25 @@ public function loadSnippetFilesIntoCollection(SnippetFileCollection $snippetFil
         $this->loadAppSnippets($snippetFileCollection);
     }
 
+    public function reset(): void
+    {
+        $this->authors = [];
+    }
+
     private function loadPluginSnippets(SnippetFileCollection $snippetFileCollection): void
     {
+        try {
+            /** @var array<string, string> $authors */
+            $authors = $this->connection->fetchAllKeyValue('
```

Comment:
can you load the authors in `loadSnippetFilesIntoCollection` and pass it as a new argument to loadPluginSnippets? 
In that way we have just a local variable and don't need to have any state 😅 . or would that not help here?

Reply:
Now passed as an argument.